src/org/exist/xquery/functions/util/Parse.java

```diff
@@ -49,17 +49,22 @@
 import java.io.StringReader;
 import java.util.Optional;
 
+
 public class Parse extends BasicFunction {
-	
+
     private static final FunctionReturnSequenceType RESULT_TYPE = new FunctionReturnSequenceType( Type.DOCUMENT, Cardinality.ZERO_OR_ONE, "the XML fragment parsed from the string" );
 
 	private static final FunctionParameterSequenceType TO_BE_PARSED_PARAMETER = new FunctionParameterSequenceType( "to-be-parsed", Type.STRING, Cardinality.ZERO_OR_ONE, "The string to be parsed" );
 
 	private static final Logger logger = LogManager.getLogger(Parse.class);
-
+	/**
+		* @deprecated use parse-xml instead
+		*/		
     public final static FunctionSignature signatures[] = {
         new FunctionSignature(
             new QName( "parse", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+						"!This function will be deprecated in a future version!" +
+						"use parse-xml instead." +
             "Parses the passed string value into an XML fragment. The string has to be " +
             "well-formed XML. An empty sequence is returned if the argument is an " +
             "empty string or sequence. If the XML is not well-formed, the function throws an " +
```

Comment:
Hi @duncdrum I like what you are doing...

If you want to deprecate an XQuery function which is implemented in Java though, there is a better way than manually changing the description. The steps to follow are:

1. The `@Deprecated` java annotation should be placed on the entire class (if there is a java class to xquery function one to one equivalence). e.g. https://github.com/eXist-db/exist/blob/develop/src/org/exist/xquery/functions/xmldb/XMLDBChmodCollection.java#L48

2. You can reference the deprecated function as the last argument of the function signature. e.g. https://github.com/eXist-db/exist/blob/develop/src/org/exist/xquery/functions/xmldb/XMLDBChmodCollection.java#L60

The example code links above are for `xmldb:chmod-collection` which is *deprecated by* `security:chmod`.

Reply:
thx for the hint adam, my problem is actually with how to do  2. would that be `ParsingFunctions.xml-parse` ? I m unclear about the signature [here](https://github.com/eXist-db/exist/blob/21f75b7f0e7ba6f3025bc0b65cd94e40d5882245/src/org/exist/xquery/functions/fn/ParsingFunctions.java#L66)